libs/components/theme/src/lib/viewport/viewport.service.ts

```diff
@@ -157,10 +162,9 @@ export class SkyAppViewportService {
         return true;
       }
 
-      // Check if the item's midpoint is visible if its space were not reserved.
-      const threshold = args.active
-        ? this.#reservedSpaces[args.position] - args.size
-        : this.#reservedSpaces[args.position];
+      // Check if the item's midpoint is visible based on current reserved spaces.
+      // Assumes reserved spaces are intended to be additive and not overlapping.
+      const threshold = reservedSpaces[args.position];
```

Comment:
This is the fix. If the space reserved after an element was greater than the height of the element, it was considered hidden.

If we assume element space is reserved in the order they appear, this change fixes that error.